feat(minecraft): update to 1.21.5 (it released!!!!)

DataM0del opened the pull request:
Still has errors.

DataM0del committed:
feat: change version to 25w06a / 1.21.5

probably will need to fix some stuff too

DataM0del committed:
fix(liquidbounce.accesswidener): comment out removed / renamed fields

DataM0del committed:
fix(CommandItemStack): comment out creative mode check because I don't know what hasCreativeInventory is in 1.21.5

DataM0del committed:
refactor(CommandConfig): fix hover & click event errors

DataM0del committed:
fix(CommandContainers): parse -> readCompound (please work)

DataM0del committed:
fix(CommandDebug): open URL debug info error

DataM0del committed:
fix(CommandFriend): hover & click event

DataM0del committed:
fix(CommandHelp): more chat events

DataM0del committed:
fix(CommandClientIntegrationSubcommand): more chat events

DataM0del committed:
fix(CommandUsername): more chat events

DataM0del committed:
fix(CommandItemEnchant#creativeOrThrow): check if we're in creative mode as a "hack fix"

DataM0del committed:
fix(CommandItemGive, CommandItemRename, CommandItemSkull): replace `hasCreativeInventory` with more hack fixes

DataM0del committed:
fix(FakePlayer): prev{Thing} -> last{Thing}

DataM0del committed:
fix(PosPoseSnapshot): prev{Thing} -> last{Thing}

DataM0del commented:
They renamed SwordItem to something else :skull:
why???

DataM0del commented:
They probably removed it because datapack moment

DataM0del commented:
~~I wonder if meteor client is updating to 1.21.5, maybe I can look at that~~edit: nope

DataM0del commented:
Ok, I found the way, just use item tags :skull:

DataM0del committed:
fix(AutoClicker#isWeaponSelected): use ItemTags for checking if an item is a sword

DataM0del committed:
todo(AutoWeapon): find `blockedByShield`

DataM0del committed:
fix(ModuleSuperKnockback): errors

DataM0del commented on src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleAutoWeapon.kt:
```diff
@@ -135,8 +135,8 @@
-                when {
-                    !isOlderThanOrEqual1_8 && target?.blockedByShield(world.damageSources.playerAttack(player)) == true
+                when { // TODO: find `blockedByShield`
+                    !isOlderThanOrEqual1_8 /*&& target?.blockedByShield(world.damageSources.playerAttack(player)) == true*/
```
ok

DataM0del committed:
fix(ModuleAutoWeapon): detekt

DataM0del committed:
fix(ModuleSwordBlock): use item tags for detecting sword items

screw your datapacks darn it!!!

DataM0del committed:
fix(ArmorEvaluation): slotType is a variable now

DataM0del commented on src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/autoarmor/ArmorEvaluation.kt:
```diff
@@ -45,15 +45,12 @@
-                else -> null
-            }
-        }.groupBy(ArmorPiece::slotType)
+            if (slot.itemStack.isIn(ItemTags.TRIMMABLE_ARMOR))
```
:|

DataM0del committed:
fix(ArmorEvaluation): detekt

DataM0del commented:
going back to look at the mc source because I can't find how to fix armor piece

DataM0del commented:
there's no way they removed the mining tool item class...

DataM0del commented:
yep, they did, datapack garbage... going to fix other errors now.

DataM0del committed:
fix(ArmorPiece, DisablerVanillaSpeed, SpecialItemExtension, SubmoduleIdPredict): errors

DataM0del committed:
fix: find & replace `getTickDelta` with `getTickProgress`

DataM0del committed:
fix(DisablerVulcanRiptide): `player.inventory.offHand[0]` -> `player.offHandStack`

DataM0del committed:
fix(ModuleInventoryTracker): `player.inventory.main` -> `player.inventory.mainStacks`

DataM0del committed:
fix(DisablerVerusExperimental): `player.lastBaseY` -> `player.lastY` (please work)

DataM0del committed:
fix(ModuleTeams): `getArmorStack` -> `getStack`

DataM0del committed:
fix(CustomAntiBotMode): most errors

DataM0del committed:
fix(MatrixAntiBotMode): all errors, probably broke something

DataM0del committed:
fix(ModuleDebugRecorder): click and hover event errors

DataM0del committed:
fix(ModuleNoClip#handleGameTick): `fallDistance` was changed to a double (lol)

DataM0del committed:
fix(ModuleSprint#shouldPreventSprint): use movementInput thing

DataM0del committed:
fix(FlyFireball, FlyGeneric, FlyNcpClip, LiquidWalkVulcan291, NoCheatPlusBow, SpeedGrimCollide): errors

DataM0del committed:
fix(SpeedVulcan286): goingSideways getter

DataM0del committed:
fix(SpeedVulcanGround286): isMovingSideways

DataM0del committed:
fix(AutoShopInventoryManager): errors

DataM0del committed:
todo(ItemCategorization#getItemFacets): fix these

DataM0del committed:
fix(ModuleOffhand, NoFallHypixelPacket, NoFallPacketJump, NoFallVerus, NoFallVulcan, WeaponItemFacet): errors

DataM0del committed:
fix(liquidbounce.accesswidener): uncomment & fix compute fall damage access widener

DataM0del committed:
fix(WeaponItemFacet): remove unused SwordItem import

DataM0del commented:
Seems like they replaced BufferRenderer with drawing with a `VertexBuffer`.
1.21.4 (WorldBorderRendering.java#render):
```java
if ((lv3 = lv2.endNullable()) != null) {
            BufferRenderer.drawWithGlobalProgram(lv3);
        }
```
...now in 1.21.5:
```java
for (WorldBorder.class_10763 lv2 : border.method_67544(k, l)) {
            if (!(lv2.distance() < d)) continue;
            this.field_56591[lv2.direction().getHorizontalQuarterTurns()].draw(lv);
        }
```

DataM0del committed:
fix(ModuleBreadcrumbs): use VertexBuffer to draw, not sure if the GlUsage value is right

DataM0del committed:
fix(ArmorPiece): fix 1 error lol

DataM0del committed:
fix(Slots): `player.inventory.mainHandStack` -> `player.mainHandStack`

DataM0del committed:
fix(ModuleMobOwners): `entity.ownerUuid` -> `entity.owner?.uuid`

DataM0del committed:
fix(PolyglotScript, RotationExtensions): more chat events & `prev{Pitch,Yaw}` -> `last{Pitch,Yaw}`

DataM0del committed:
fix(RunnableClickEvent): add getAction (returns `RUN_COMMAND` action) function & remove superclass params

DataM0del committed:
fix(EntityExtensions): moving property & `prev{Name}` -> `last{Name}` for some things

DataM0del committed:
fix(InventoryUtils): errors

DataM0del committed:
fix(MixinClientWorld, MixinInput): got bored with fixing Kotlin errors, going on to mixins...

DataM0del committed:
fix(MixinClientPlayerEntity): 4 more errors in this file

DataM0del committed:
chore: update yarn

DataM0del committed:
fix(ModuleSurround): prev{X,Y,Z} -> last{X,Y,Z}

DataM0del committed:
fix(MurderMysterySwordDetection#isSword): detect swords and pickaxes with item tags

DataM0del commented:
only 99 errors left :money_mouth_face:

DataM0del committed:
fix(MixinClientPlayerEntity#hookCustomMultipler): please work

DataM0del committed:
fix(ModuleHoleFiller): prev{X,Y,Z} -> last{X,Y,Z}

ccetl commented:
> Seems like they replaced BufferRenderer with drawing with a `VertexBuffer`. 1.21.4 (WorldBorderRendering.java#render):
> 
> ```java
> if ((lv3 = lv2.endNullable()) != null) {
>             BufferRenderer.drawWithGlobalProgram(lv3);
>         }
> ```
> 
> ...now in 1.21.5:
> 
> ```java
> for (WorldBorder.class_10763 lv2 : border.method_67544(k, l)) {
>             if (!(lv2.distance() < d)) continue;
>             this.field_56591[lv2.direction().getHorizontalQuarterTurns()].draw(lv);
>         }
> ```

~~No it's the same. The code regarding that did not change.~~
Okay i was on an outdated snapshot, it did change. But please reuse the VertexBuffers.

ccetl commented on src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/CommandConfig.kt:
```diff
@@ -91,14 +92,12 @@ object CommandConfig : CommandFactory {
                                 .withClickEvent(
-                                    ClickEvent(
-                                        ClickEvent.Action.SUGGEST_COMMAND,
+                                    CopyToClipboard(
```
~~This is not the same.~~
It is but please use `ClickEvent.CopyToClipboard` to make it clear and be consistent.

DataM0del replied:
Ok, probably should also make it actually the same since suggest command != copy to clipboard

ccetl commented on src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/ingame/creative/CommandItemEnchant.kt:
```diff
@@ -181,7 +181,8 @@ object CommandItemEnchant : CommandFactory, MinecraftShortcuts {
     private fun creativeOrThrow(command: Command) {
-        if (mc.interactionManager?.hasCreativeInventory() == false) {
+        // TODO: you can probably have creative inventory in survival through abilities or something, "hack fix" I guess
+        if (mc.interactionManager?.gameMode?.isCreative == false) {
```
```suggestion
        if (!player.isInCreativeMode) {
```
Used by minecraft in the parts that previously used the function. 

No need to check if the player is ingame as this is an ingame command.

ccetl commented on src/main/kotlin/net/ccbluex/liquidbounce/utils/client/RunnableClickEvent.kt:
```diff
@@ -24,10 +24,15 @@ import net.minecraft.text.MutableText
 
+    // TODO: lol
+    override fun getAction(): ClickEvent.Action? {
+        return ClickEvent.Action.RUN_COMMAND
```
No, should have the same behavior as before.

DataM0del replied:
You have to override this otherwise it will error. What else am I going to do that won't cause an error?

ccetl replied:
Return `null`. And verify the logic, but i might do that myself later.

DataM0del replied:
> Return `null`. And verify the logic, but i might do that myself later.

Oh. :skull:
hmm why `?` at the end of the return type

DataM0del replied:
Okay, I'm going to try that out... After I get this to run since there's no way I can verify this without it building.
Just hope that it doesn't null pointer exception :skull:

DataM0del committed:
fix(ModuleProjectileAimbot): added handItems

literally just taken from 1.21.4 source, it just returns a List of the main hand item and the offhand item

DataM0del committed:
fix: use player.isInCreativeMode instead of isCreative in interaction manager

Co-authored-by: ccetl <109693935+ccetl@users.noreply.github.com>

DataM0del committed:
fix(CommandConfig): use SuggestCommand instead of CopyToClipboard (oops)

DataM0del committed:
fix(MurderMysterySwordDetection): ok detekt

DataM0del committed:
fix: import handItems extension where handItems is used

DataM0del committed:
fix(ContainerFunctions): check for creative mode instead of hasCreativeInventory

DataM0del committed:
refactor(ModuleBreadcrumbs): reuse vertex buffer

DataM0del committed:
fix(RenderBufferBuilder#draw): use vertex buffer

DataM0del committed:
Merge branch 'CCBlueX:nextgen' into feat/minecraft/1.21.5

DataM0del commented:
update never

DataM0del committed:
Merge branch 'nextgen' into feat/minecraft/1.21.5

DataM0del committed:
chore(gradle.properties): update to latest snapshot

DataM0del committed:
refactor(BlockExtensions): hasCreativeInventory() -> gameMode.isCreative()

DataM0del committed:
Revert "chore(gradle.properties): update to latest snapshot"

This reverts commit 9ec882bde9f9f73649c639679245f06e9781fd51.

DataM0del committed:
fix: use Gradle 8.12

1zun4 commented:
When Minecraft 1.21.5 is released, I will remember this as a reference in case I get lost in the changes. Thanks @DataM0del - even if you're now banned from the repository.

DataM0del commented:
> When Minecraft 1.21.5 is released, I will remember this as a reference in case I get lost in the changes. Thanks @DataM0del - even if you're now banned from the repository.

I can't reopen this PR because you closed it, and GitHub won't let you reopen a PR if someone else closed it.
